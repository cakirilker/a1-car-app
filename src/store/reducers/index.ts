export * from './filters.reducer';
export * from './cars.reducer';
export * from './root.reducer';
