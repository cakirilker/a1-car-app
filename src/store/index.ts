import { ThunkAction } from 'redux-thunk';
import { RootReducer, RootState } from './reducers';
import api from '../api';
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import { Action } from 'redux';

const store = configureStore({
  reducer: RootReducer,
  middleware: getDefaultMiddleware({
    thunk: { extraArgument: { api } },
  }),
});

export type AppDispatch = typeof store.dispatch;
export type AppThunk = ThunkAction<void, RootState, unknown, Action<string>>;

export default store;
