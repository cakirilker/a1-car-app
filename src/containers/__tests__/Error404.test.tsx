import React from 'react';
import {
  render,
  RenderOptions,
  RenderResult,
  screen,
  fireEvent,
} from '@testing-library/react';
import { Error404 } from '..';
import { MemoryRouter, Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import store from '../../store';

const AllTheProviders = ({ children }: any) => {
  return (
    <MemoryRouter>
      <Provider store={store}>{children}</Provider>
    </MemoryRouter>
  );
};

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

describe('404 Page Component', () => {
  let context: RenderResult;
  beforeEach(() => {
    context = customRender(<Error404 />);
  });
  test('should render correctly', () => {
    expect(context).toMatchSnapshot();
  });

  test('should display not found text', () => {
    expect(screen.getByRole('heading', { level: 3 })).toHaveTextContent(
      '404 - Not Found',
    );
  });

  test('should display informative message', () => {
    expect(screen.getByRole('heading', { level: 5 })).toHaveTextContent(
      'Sorry, the page you are looking for does not exists.',
    );
  });

  test('should display return to the homepage text', () => {
    expect(screen.getByRole('heading', { level: 6 })).toHaveTextContent(
      'You can always go back to the homepage.',
    );
  });

  test('should redirect to the homepage when link is clicked', () => {
    context.unmount();
    const history = createMemoryHistory();
    history.push('/some-route');
    render(
      <Router history={history}>
        <Provider store={store}>
          <Error404 />
        </Provider>
      </Router>,
    );
    const link = screen.getByTestId('homepage-link');
    fireEvent.click(link);
    expect(history.location.pathname).toEqual('/');
  });
});
