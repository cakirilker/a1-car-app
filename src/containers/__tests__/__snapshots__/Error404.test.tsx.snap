// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 Page Component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root makeStyles-root-1 MuiContainer-maxWidthSm"
      >
        <img
          alt="Auto1 Logo"
          class="makeStyles-image-2"
          src="logo.png"
        />
        <h3
          class="MuiTypography-root makeStyles-header-3 MuiTypography-h3 MuiTypography-gutterBottom"
        >
          404 - Not Found
        </h3>
        <h5
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom"
        >
          Sorry, the page you are looking for does not exists.
        </h5>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom"
        >
          You can always go back to the
           
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
            data-testid="homepage-link"
            href="/"
          >
            homepage
          </a>
          .
        </h6>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root makeStyles-root-1 MuiContainer-maxWidthSm"
    >
      <img
        alt="Auto1 Logo"
        class="makeStyles-image-2"
        src="logo.png"
      />
      <h3
        class="MuiTypography-root makeStyles-header-3 MuiTypography-h3 MuiTypography-gutterBottom"
      >
        404 - Not Found
      </h3>
      <h5
        class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom"
      >
        Sorry, the page you are looking for does not exists.
      </h5>
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom"
      >
        You can always go back to the
         
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
          data-testid="homepage-link"
          href="/"
        >
          homepage
        </a>
        .
      </h6>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
